faker/providers/misc/__init__.py

```diff
@@ -130,12 +130,17 @@ def locale(self):
     def language_code(self):
         return self.random_element(Provider.language_locale_codes.keys())
 
-    def uuid4(self):
+    def uuid4(self, cast_to=str):
```

Comment:
I find the use of `cast_to` a (tiny) bit confusing in case of `None`. Consider

``` python
fake.uuid4(cast_to=str)
fake.uuid4(cast_to=None)
```
The first one reads as "cast to string", and the second reads as "cast to none", but I don't want my uuid to be `None`!

What if it's up to the user to provide an identity function when they want a plain UUID?

``` python
fake.uuid4(cast_to=lambda x: x)
```

Reply:
Hi agree with you @ThibaultLemaire , let's remove the special case for `None`.